src/main/java/edu/indiana/dlib/amppd/service/impl/PreprocessServiceImpl.java

```diff
@@ -176,9 +182,15 @@ public String retrieveMediaInfo(String filepath) {
 	 */
 	@Override	
 	public Asset retrieveMediaInfo(Asset asset) {
-		String mediaInfo = retrieveMediaInfo(asset.getPathname());
+		String filepath = asset.getPathname();
+		return retrieveMediaInfo(asset, filepath);
+	}
+	
+	@Override	
+	public Asset retrieveMediaInfo(Asset asset, String filepath) {
+		String mediaInfo = retrieveMediaInfo(filepath);
 		if (StringUtils.isEmpty(mediaInfo)) {
-			throw new PreprocessException("Error retrieving media info for Asset " + asset.getId() + ": the result is empty");
+			throw new PreprocessException("Error retrieving media info for " + filepath + ": the result is empty");
 		}
 
```

Comment:
to keep it simple, I'd prefer not having to pass in an extra param for filepath if possible - for ex, you can set the filepath in primaryfile to point to the actual file before preprocess, and update filepath later when moving file. The Asset id might be null if it has never been saved to DB; but can we utilize whatever other info (such as label etc) to print to log? These fields are populated before preprocessing right?

Alternatively, one could also catch/rethrow exception one level up (in batch service) and log whatever context info batch ingest knows of (row number for ex). 
Also, the